src/Workspaces/Core/Desktop/Workspace/MSBuild/MSBuildProjectLoader.cs

```diff
@@ -97,25 +97,22 @@ public void AssociateFileExtensionWithLanguage(string projectFileExtension, stri
 
         private void SetSolutionProperties(string solutionFilePath)
         {
+            // When MSBuild is building an individual project, it doesn't define $(SolutionDir).
+            // However when building an .sln file, or when working inside Visual Studio,
+            // $(SolutionDir) is defined to be the directory where the .sln file is located.
+            // Some projects out there rely on $(SolutionDir) being set (although the best practice is to
+            // use MSBuildProjectDirectory which is always defined).
             if (!string.IsNullOrEmpty(solutionFilePath))
             {
-                // When MSBuild is building an individual project, it doesn't define $(SolutionDir).
-                // However when building an .sln file, or when working inside Visual Studio,
-                // $(SolutionDir) is defined to be the directory where the .sln file is located.
-                // Some projects out there rely on $(SolutionDir) being set (although the best practice is to
-                // use MSBuildProjectDirectory which is always defined).
-                if (!string.IsNullOrEmpty(solutionFilePath))
+                string solutionDirectory = Path.GetDirectoryName(solutionFilePath);
+                if (!solutionDirectory.EndsWith(@"\", StringComparison.Ordinal))
                 {
-                    string solutionDirectory = Path.GetDirectoryName(solutionFilePath);
-                    if (!solutionDirectory.EndsWith(@"\", StringComparison.Ordinal))
-                    {
-                        solutionDirectory += @"\";
-                    }
+                    solutionDirectory += @"\";
+                }
 
-                    if (Directory.Exists(solutionDirectory))
-                    {
-                        _properties = _properties.SetItem(SolutionDirProperty, solutionDirectory);
-                    }
+                if (Directory.Exists(solutionDirectory))
```

Comment:
Isn't the solution file at this point known to exist? (Aren't we already loading it?) Can this ever fail?

(yes, I know this logic was there, but if we're deleting stupid code we mind as well delete more stupid code)